src/main/scala/org/ergoplatform/nodeView/wallet/WalletScanLogic.scala

```diff
@@ -115,10 +115,16 @@ object WalletScanLogic extends ScorexLogging {
     }
 
     // Resolve boxes related to mining income
-    val resolvedBoxes = registry.unspentBoxes(MiningScanId).flatMap { tb =>
-      val spendable = resolve(tb.box, walletVars.proverOpt, stateContext, height)
-      if (spendable) Some(tb.copy(scans = Set(PaymentsScanId))) else None
-    }
+    // We take only 5 oldest boxes to avoid long scans in the testnet where one wallet
+    // (corresponding to a node generating almost all the blocks) can possess
+    // hundreds of unspent mining reward boxes.
+    val miningBoxes = registry.unspentBoxes(MiningScanId).sortBy(_.inclusionHeightOpt.getOrElse(0)).take(5)
```

Comment:
Please clarify in the code what it means to take `5 oldest` and why it is safe (make sence). 
Note, before the `take` method comes into play, all set of boxes is returned (as a Seq with underlying ListBuffer probably) and then is it sorted.

Reply:
As node settings available here, I made a better approach, choosing boxes which are mature enough specifically.